### ISSUE TITLE
support for OAuth based logins.

### ISSUE THREAD

**Issue by superfell** [superfell/SoqlX#5]

**Comment by ckoppelman** [comment]
Now that `soql://` urls are supported, can that be used for oath authentication?

**Comment by StevenPanSideInc** [comment]
Hi, running into the same issue with SSO based logins - wanted to see if theres any updates to this

**Comment by abarash** [comment]
+1 from me - we're rolling out more enhanced security to our whole org and are trying to force SSO, but that blocks our folks who use SoqlX regularly from being able to use it. Thanks for a great tool!

**Comment by kensation** [comment]
@superfell Are there any updates on whether SoqlX will work with SSO.  Also - Will SoqlX still work with the MFA requirement Salesforce will be requiring February 1, 2022?

**Comment by superfell** [comment]
SoqlX should continue to work, according to the MFA FAQ it doesn't affect API logins

https://help.salesforce.com/s/articleView?id=000352937&type=1

THere's no change WRT to OAuth based logins not being implemented

**Comment by kensation** [comment]
Cool. Thanks for the quick response!

**Comment by superfell** [comment]
Once OAuth logins are added, is the API login approach still needed? I can't think of any compelling reasons.

**Comment by kensation** [comment]
> Once OAuth logins are added, is the API login approach still needed? I can't think of any compelling reasons.

I am not sure. I would pose my original question though. If you end up going OAuth only, how will that work with SSO/MFA? Does the OAuth endpoint handle all that on its end per the settings set up in Salesforce, and just grants the application access? I am not familiar enough with how the OAuth  login flow works with SSO/MFA.

**Comment by superfell** [comment]
In the OAuth flow, SoqlX will open a specific URL in the users default browser, they will do the standard salesforce.com UI login operation, including any org specific SSO/MFA setup and once logged in, will be presented with a dialog to authorize SoqlX. If authorized, the browser passes the relevant details back to the SoqlX which include a api session token that SoqlX can then use.

<img width="461" alt="Screen Shot 2021-09-24 at 11 18 21 AM" src="https://user-images.githubusercontent.com/399098/134722243-6e88c2d1-b5a4-4165-98aa-9326e31e0f2f.png">

**Comment by superfell** [comment]
I put together a beta version that uses OAuth rather than the API based login. Give it a go and let me know what you think.
https://superfell.com/osx/soqlx/SoqlXplorer_v4.2_b1.zip

**Comment by ckoppelman** [comment]
It works great for me!  I can do my high-level auth DML in SoqlX without a problem!

(UX nitpick:)

> The gears in Saved Logins and domains list are a little hard to parse (especially based on where they are).  Many iPhone apps use "Edit" for this purpose, and "+" to add a new account, both at the top of the screen.  Edit also lets you drag logins higher or lower.  I wonder if that sort of UX could be useful here.
> 
> If not, putting "gear" into words and placing it among the buttons would be helpful.

**Comment by superfell** [comment]
Thanks for the UI feedback, I've been through a few iterations but still don't like it. I'll tinker with it some more.
